src/main/kotlin/it/pagopa/ecommerce/eventdispatcher/queues/v2/TransactionAuthorizationRequestedQueueConsumer.kt

```diff
@@ -1,85 +1,35 @@
 package it.pagopa.ecommerce.eventdispatcher.queues.v2
 
 import com.azure.spring.messaging.checkpoint.Checkpointer
+import io.vavr.control.Either
 import it.pagopa.ecommerce.commons.documents.v2.*
-import it.pagopa.ecommerce.commons.domain.v2.EmptyTransaction
-import it.pagopa.ecommerce.commons.domain.v2.Transaction
-import it.pagopa.ecommerce.commons.domain.v2.pojos.BaseTransaction
-import it.pagopa.ecommerce.commons.domain.v2.pojos.BaseTransactionWithRequestedAuthorization
-import it.pagopa.ecommerce.commons.generated.server.model.TransactionStatusDto
 import it.pagopa.ecommerce.commons.queues.QueueEvent
-import it.pagopa.ecommerce.commons.queues.StrictJsonSerializerProvider
-import it.pagopa.ecommerce.commons.queues.TracingUtils
-import it.pagopa.ecommerce.eventdispatcher.client.TransactionsServiceClient
-import it.pagopa.ecommerce.eventdispatcher.repositories.TransactionsEventStoreRepository
-import it.pagopa.ecommerce.eventdispatcher.services.eventretry.v2.AuthorizationStateRetrieverRetryService
-import it.pagopa.ecommerce.eventdispatcher.services.v2.NpgStateService
-import it.pagopa.ecommerce.eventdispatcher.utils.DeadLetterTracedQueueAsyncClient
-import org.slf4j.Logger
-import org.slf4j.LoggerFactory
+import it.pagopa.ecommerce.eventdispatcher.queues.v2.helpers.AuthorizationRequestedEvent
+import it.pagopa.ecommerce.eventdispatcher.queues.v2.helpers.AuthorizationRequestedHelper
 import org.springframework.beans.factory.annotation.Autowired
 import org.springframework.stereotype.Service
 import reactor.core.publisher.Mono
-import reactor.kotlin.core.publisher.switchIfEmpty
 
 /**
  * Event consumer for events related to refund retry. This consumer's responsibilities are to handle
  * refund process retry for a given transaction
  */
 @Service("TransactionAuthorizationRequestedQueueConsumerV2")
 class TransactionAuthorizationRequestedQueueConsumer(
-  @Autowired private val transactionsServiceClient: TransactionsServiceClient,
-  @Autowired private val transactionsEventStoreRepository: TransactionsEventStoreRepository<Any>,
-  @Autowired
-  private val authorizationStateRetrieverRetryService: AuthorizationStateRetrieverRetryService,
-  @Autowired private val npgStateService: NpgStateService,
-  @Autowired private val deadLetterTracedQueueAsyncClient: DeadLetterTracedQueueAsyncClient,
-  @Autowired private val tracingUtils: TracingUtils,
-  @Autowired private val strictSerializerProviderV2: StrictJsonSerializerProvider
+  @Autowired private val authorizationRequestedHelper: AuthorizationRequestedHelper,
 ) {
 
-  var logger: Logger =
-    LoggerFactory.getLogger(TransactionAuthorizationRequestedQueueConsumer::class.java)
-
   fun messageReceiver(
-    parsedEvent: QueueEvent<TransactionAuthorizationRequestedEvent>,
+    parsedEvent:
+      Either<
+        QueueEvent<TransactionAuthorizationRequestedEvent>,
+        QueueEvent<TransactionAuthorizationOutcomeWaitingEvent>>,
```

Comment:
Change this fun input to receive only `TransactionAuthorizationRequestedEvent` and then wrap it to Either left before pass it to the centralized logic. This way this handler can only receive the TransactionAuthorizationRequestedEvent